src/components/ResoruceIcon.tsx

```diff
@@ -0,0 +1,47 @@
+import { Icon } from "@canonical/react-components";
+import { FC } from "react";
+
+export type ResourceIconType =
+  | "container"
+  | "virtual-machine"
+  | "snapshot"
+  | "profile"
+  | "project"
+  | "cluster-member"
+  | "network"
+  | "pool"
+  | "volume"
+  | "iso-volume"
+  | "image"
+  | "oidc-identity"
+  | "certificate"
+  | "auth-group"
+  | "device";
+
+export const resourceIcons: Record<ResourceIconType, string> = {
+  container: "profiles",
+  "virtual-machine": "pods",
+  snapshot: "snapshot",
+  profile: "repository",
+  project: "folder",
+  "cluster-member": "single-host",
+  network: "exposed",
+  pool: "status-queued-small",
+  volume: "status-queued-small",
+  "iso-volume": "iso",
+  image: "image",
+  "oidc-identity": "user",
+  certificate: "certificate",
+  "auth-group": "user-group",
+  device: "units",
+};
+
+interface Props {
+  type: ResourceIconType;
+}
+
+const ResoruceIcon: FC<Props> = ({ type }) => {
+  return <Icon name={resourceIcons[type]} />;
```

Comment:
Don't we need to render null if the lookup fails?

Reply:
I was doing it previously, but it shouldn't be necessary since we have strict type check for the `type` prop. Not too long ago, we implemented `tsc` for the `lint-js` script. So CI will fail if someone ignored the type error message.